### DISCUSSION TITLE
Versioning of the file layout

### DISCUSSION THREAD

**Discussion by jonberkery-work** [CMSgov/price-transparency-guide#241] [Q&A] [answered]
What is the plan for the versioning of this file layout?  From the CMS meeting on 10.20.2021, it was mentioned that version 1.0.0 will not become final until March 2022.

Knowing that, my questions are ....

1. Will each major version go through a lifecycle?  For Example - Emerging, Approved, Recommended, Declining, Prohibited
2. If a lifecycle is defined, how long does a version remain at each stage?
3. How many major versions will remain active\approved at a given time?
4. What is the required timeframe to migrate to an approved major version?
5. Does a "file producer" have to migrate to minor versions?
6. Can a "file producer" skip a major version?
7. Will versions be backwards compatible?

**Comment by shaselton-usds** [comment, marked as answer]
It was mentioned that there'll be a light gitflow process -- with GH Actions in place, development will soon switch over to a "develop" branch in which a "release" branch will be created eventually signaling a stabilized v1.0 on the horizon to be merged back into "master".

At this point, v1.0 will be the only major version. If significant changes are made to the schema, that'll usually be following a policy (NPRM) change and that would probably look like a v2.0.

The March 1st, 2022 date is the latest date in which a v1.0 could be determined. If the schema has stabilized beforehand, v1.0 can be released earlier.

After v1.0 has been determined, that is what compliance will based on. Of course the schema can still go through iterations and would trigger a v1.1 or v1.2. "File producers" wouldn't have to incorporate these changes, but some of these change could be quite beneficial in terms of file size optimizations, so it might make sense to keep migrating with the versions.

Usually major version bumps signal breaking changes. Anything after v1.0 will not have breaking changes (or it shouldn't). Again, major changes to the schema would most likely be due to major policy changes which would be found in an NPRM.
